src/EditorFeatures/CSharpTest/UseExpressionBody/UseExpressionBodyForMethodsTests.cs

```diff
@@ -277,5 +277,73 @@ void Foo()
     void Foo() => throw Bar(); // Comment
 }", options: UseExpressionBody, compareTokens: false);
         }
+
+        [WorkItem(17120, "https://github.com/dotnet/roslyn/issues/17120")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseExpressionBody)]
+        public async Task TestDirectives1()
+        {
+            await TestAsync(
+@"
+#define DEBUG
+using System;
+
+class Program
+{
+    void Method()
+    {
+#if DEBUG
+        [|Console|].WriteLine();
+#endif
+    }
+}",
+@"
+#define DEBUG
+using System;
+
+class Program
+{
+    void Method() =>
```

Comment:
This isn't really helping the user that much, since this will fail to parse when `#if DEBUG` is false :-/